include/bredis/impl/connection.ipp

```diff
@@ -0,0 +1,172 @@
+//
+//
+// Copyright (c) 2017 Ivan Baidakou (basiliscos) (the dot dmol at gmail dot com)
+//
+// Distributed under the MIT Software License
+//
+#pragma once
+
+#include "common.ipp"
+#include <algorithm>
+#include <cassert>
+#include <ostream>
+#include <type_traits>
+
+namespace bredis {
+
+template <typename NextLayer>
+template <typename WriteCallback, typename DynamicBuffer>
+BOOST_ASIO_INITFN_RESULT_TYPE(WriteCallback,
+                              void(boost::system::error_code, std::size_t))
+Connection<NextLayer>::async_write(DynamicBuffer &tx_buff,
+                                   const command_wrapper_t &command,
+                                   WriteCallback write_callback) {
+    namespace asio = boost::asio;
+    namespace sys = boost::system;
+    using boost::asio::async_write;
+    using Signature = void(boost::system::error_code, std::size_t);
+    using real_handler_t =
+        typename asio::handler_type<WriteCallback, Signature>::type;
+
+    std::ostream os(&tx_buff);
+    auto string = boost::apply_visitor(command_serializer_visitor(), command);
+    os.write(string.c_str(), string.size());
+    tx_buff.commit(string.size());
+
+    real_handler_t handler(std::forward<WriteCallback>(write_callback));
+    return async_write(stream_, tx_buff, handler);
```

Comment:
There is no need of it, as the `WriteCallbacksignature` matches to the signature of `asyn_write` of Boost::ASIO